src/lists.py

```diff
@@ -39,20 +41,30 @@ def print_list(x):
     print("Incorrect number of arguments.", file=sys.stderr)
     sys.exit(1)
 
-match sys.argv[1]:
+match sys.argv[1]: # sys.argv[1] is a list. 
     case "mean":
         # put your solution to the first exercise here
-        mean = "mean of x"
+        total = 0
+        length = 0
+        for n in sys.argv[1]:
```

Comment:
You are iterating through the first argument (which in this case must be `"mean"`). That is a string, but you are trying to use it as a list of integers. You want to iterate through the list `x` instead (that we read from `stdin` at the top of the script)